plugin/al/lib/AL_USDMaya/AL/usdmaya/cmds/LayerCommands.cpp

```diff
@@ -116,16 +116,6 @@ MSyntax LayerCommandBase::setUpCommonSyntax()
   return syntax;
 }
 
-//----------------------------------------------------------------------------------------------------------------------
-MArgDatabase LayerCommandBase::makeDatabase(const MArgList& args)
-{
-  MStatus status;
-  MArgDatabase database(syntax(), args, &status);
-  if(!status)
-    throw status;
-  return database;
-}
-
 //----------------------------------------------------------------------------------------------------------------------
```

Comment:
So I had to make this change because it was causing quite a few crashes when running the tests.  Unfortunately it’s hitting a problem that needs to be fixed in Maya itself.  Not sure how it’s gone this long without being discovered, but Maya isn’t dealing with the memory it allocates correctly when using the MArgDatabase constructor that we are using here, and the copy of that class in the return of the utility function ends up causing a crash when it goes out of scope.  

I think I had originally tried adding a macro in one of the test utility headers but I didn’t like how it looked.  E.g. I don’t like having a macro that defines variables for you that are then used elsewhere in the calling code.  And when you leave the variable definition local and then just add a macro that maybe checks a return value and asserts…you only end up saving a line or two of code anyways.  Since that utility function had to go given the Maya crash, and I had to visit all of these places, I just went ahead and added the code.

I really didn’t think much about the try/catch code.  In the first test I looked at it seemed like the utility function I was removing was the only thing that would throw something, so I could have also got rid of the try/catch, but I wasn’t 100% if there was anything else in the middle that could also have thrown something so I just decided to leave it in.  All the function does when it catches something is return the status though, so I think having this new chunk of code do the return is the proper thing to do.